compiler/rustc_typeck/src/check/fn_ctxt.rs

```diff
@@ -2887,10 +2887,11 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                                 self.var_for_def(span, param)
                             }
                         }
-                        GenericParamDefKind::Const => {
-                            // FIXME(const_generics:defaults)
-                            // No const parameters were provided, we have to infer them.
-                            self.var_for_def(span, param)
+                        GenericParamDefKind::Const { has_default, .. } => {
+                            if infer_args || !has_default {
+                                return self.var_for_def(span, param);
```

Comment:
This is the wrong way round.

Reply:
I manually inverted the check so I could short circuit, so I believe this is correct